src/include/storage/index/index.h

```diff
@@ -37,6 +39,17 @@ class Index {
    */
   const IndexMetadata metadata_;
 
+  /**
+   * Determine if a tuple is visible by asking the DataTable associated with the TupleSlot. Used for scans.
+   * @param txn the calling transaction
+   * @param slot the slot of the tuple to check visibility on
+   * @return true if tuple is visible to this txn, false otherwise
+   */
+  static bool IsVisible(const transaction::TransactionContext &txn, const TupleSlot slot) {
+    const auto *const data_table = slot.GetBlock()->data_table_;
+    return data_table->IsVisible(txn, slot);
```

Comment:
We need more logic here.  This just answers is the *tuple* visible, but we actually need to ask is the *tuple with the indexed attribute I'm looking for* visible.  Specifically, updates on indexed attributes will break this logic.

Reply:
This should be good because of the table-level insert.